core/src/core/transaction.rs

```diff
@@ -852,11 +875,7 @@ impl TransactionBody {
 		};
 
 		// Verify the unverified tx kernels.
-		// No ability to batch verify these right now
-		// so just do them individually.
-		for x in &kernels {
-			x.verify()?;
-		}
+		TxKernel::batch_sig_verify(&kernels)?;
```

Comment:
I'm guessing we are fine just batching them all up like this in the context of a block.
But is there a limit to how many we can batch?

Reply:
The limit should be big enough. At least 100,000 batching is OK:
```sh
running 1 test
Verifying aggsig batch of 100000
100000 signatures verified in 6408.601ms
test aggsig::tests::test_aggsig_batch ... ok
```